modules/network/src/main/java/org/apache/ignite/network/internal/netty/HandshakeHandler.java

```diff
@@ -41,31 +44,38 @@ public HandshakeHandler(HandshakeManager manager) {
     }
 
     /** {@inheritDoc} */
-    @Override public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
+    @Override public void handlerAdded(ChannelHandlerContext ctx) {
         HandshakeAction handshakeAction = manager.init(ctx.channel());
 
         handleHandshakeAction(handshakeAction, ctx);
     }
 
     /** {@inheritDoc} */
-    @Override public void channelActive(ChannelHandlerContext ctx) throws Exception {
+    @Override public void channelActive(ChannelHandlerContext ctx) {
         HandshakeAction handshakeAction = manager.onConnectionOpen(ctx.channel());
 
+        manager.handshakeFuture().whenComplete((unused, throwable) -> {
+            if (throwable != null) {
+                LOG.error("Error when performing handshake", throwable);
+                ctx.close();
+            }
+        });
```

Comment:
```suggestion
        manager.handshakeFuture().exceptionally((throwable) -> {
            LOG.error("Error when performing handshake", throwable);
            
            ctx.close();            
        });
```

Reply:
`exceptionally` is intended to be used as a recovery from an exception (it returns a value). I think that `whenComplete` better suites this particular use case